streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -211,28 +207,44 @@ public synchronized void start() {
      * and then wait for them to join.
      *
      * @throws IllegalStateException if process has not started yet
+     * @param timeout   how long to wait for {@link StreamThread}s to shutdown
+     * @param timeUnit  unit of time used for timeout
      */
-    public synchronized void close() {
+    public synchronized void close(final long timeout, final TimeUnit timeUnit) {
```

Comment:
We do have all kind of default timeout. IMHO, having two overload would be the better choice. What about using 30 seconds as default? That's also the default for consumers to timeout?